frontend/src/components/NewAgreement/AgreementCreatedSuccessfully.tsx

```diff
@@ -1,27 +1,90 @@
-import { Box, Button, Container, Divider, Stack, Typography } from '@mui/material';
+import {
+  Box,
+  Button,
+  Container,
+  Divider,
+  IconButton,
+  Stack,
+  SvgIcon,
+  Typography,
+} from '@mui/material';
 import { FC, useContext } from 'react';
 import { useTranslation } from 'react-i18next';
-import DoneRoundedIcon from '@mui/icons-material/DoneRounded';
 import { StringBank } from 'strings';
 import { useNavigate } from 'react-router-dom';
 import { GroupContext } from 'contexts/group';
+import handLogo from 'assets/icons/hand-v@3x.png';
+import styled from 'styled-components';
+import { ReactComponent as CopyIcon } from 'assets/icons/copy.svg';
 
-const AgreementCreatedSuccessfully: FC = () => {
+const CopyIconWrapper = styled(SvgIcon)`
+  svg path {
+    fill: #adb2b8;
+  }
+`;
+
+const CopyEL: FC<{ text: string }> = ({ text }) => {
+  return (
+    <IconButton
+      sx={{ marginLeft: '2rem', marginRight: '1rem' }}
+      onClick={() => {
+        navigator.clipboard.writeText(text);
+      }}
+    >
+      <CopyIconWrapper>
+        <CopyIcon />
+      </CopyIconWrapper>
+    </IconButton>
+  );
+};
+
+const CopyAgreementContainer = styled(Stack)`
+  && {
+    width: 298px;
+    height: 42px;
+    flex-grow: 0;
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+    padding: 24px 16px;
+    border-radius: 4px;
+    border: solid 1px #565d68;
+    background-color: rgba(0, 0, 0, 0.08);
+  }
+`;
+
+const AgreementTextSpan = styled.span`
+  && {
+    width: 209px;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    overflow: hidden;
+    font-size: 14px;
+    font-weight: normal;
+    font-stretch: normal;
+    font-style: normal;
+    line-height: 1.71;
+    letter-spacing: normal;
+    text-align: left;
+    color: #adb2b8;
+  }
+`;
```

Comment:
No need for all of this styling.
Also - we would rather not to use fixed widths for those elements, as they should fit different screen sizes.
In most cases we should implement size-related behaviors of elements using [flexbox grow, shrink, basis etc.](https://css-tricks.com/snippets/css/a-guide-to-flexbox)

Reply:
removed, but it forces my to use important, since the stack style overwriddes the other nested styles